server/lib/routes/get-authorization.js

```diff
@@ -0,0 +1,66 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const url = require('url');
+
+const logger = require('../logging/log')('server.get-authorization');
+
+const ACTION_TO_PATHNAMES = {
+  'email': '',
+  'force_auth': 'force_auth',
+  'signin': 'signin',
+  'signup': 'signup'
+};
+
+function actionToPathname(action) {
+  if (action === undefined) {
+    return '';
+  }
+
+  if (ACTION_TO_PATHNAMES.hasOwnProperty(action)) {
+    return ACTION_TO_PATHNAMES[action];
+  }
+
+  throw new Error('Bad action parameter');
+}
+
+module.exports = function () {
+  return {
+    method: 'get',
+    path: '/authorization',
+    process: function (req, res) {
+      let err = false;
+      const route = url.parse('/', true);
+
+      try {
+        route.pathname += actionToPathname(req.query.action);
+      } catch (e) {
+        err = true;
+        logger.error('Invalid request parameter', req.query.action);
+        return res.send(500, 'Invalid request parameter ' + req.query.action)  ;
```

Comment:
Two comments:

1. We have been treating invalid query parameters as a 400 error instead of a 500, [5xx is for server side errors](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#5xx_Server_errors).
2. I'd like to lean on server side query parameter validation more so that we can remove a lot of the validation from the client. Instead of sending a simple text response, let's render a template that says *which* query parameter is invalid but not the value. Those values can be anything and malicious actors could try to use them as an XSS vector. Here's what we show if the client detects validation fails for an OAuth query param:

<img width="429" alt="screen shot 2018-05-21 at 13 18 38" src="https://user-images.githubusercontent.com/848085/40307261-a7d94a4e-5cf9-11e8-80dd-4607a1ee36e0.png">

Here it would be great to have a generic 400 template that we can lean on that we display if one of the query params fails validation. To make this work with [the validate comment made above](), instead of saying which template to render here, in routes.js, we'll need to [check the Accept header and if HTML is expected, render the template](https://github.com/mozilla/fxa-content-server/blob/eabe9c022d6135bd9ac10d11229b56a95ff3aa77/server/lib/routes.js#L119).

Reply:
do we already have this template ?